### PR TITLE
(BLO-61) fix di bug

### PR DESCRIPTION
**Pull request by saltz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,54 @@
...
+*container*
+```ts
+DIContainer.bind<DataAccessLayer>(DataAccessLayer).toSelf().inTransientScope();
+
+DIContainer.bind<DataSource>("DataSource").toConstantValue(DataSource.MONGO_DB);
+DIContainer.bind<IClientConfiguration>("Configuration")
+    .toConstantValue(new MongoDbConfiguration("connection string", "database"));
+```
+
+*consumer (typicaly a service)*
```
Comment (timofudala) [inline README.md]: Please change `typicaly` to `typically`

File: README.md [hunk 2]
```diff
@@ -3,3 +3,54 @@
...
+```ts
+class Main {
+    private dataAccessLayer: DataAccessLayer;
+
+    constrcutor(@inject(DataAccessLayer) dataAccessLayer: DataAccessLayer) {
```
Comment (timofudala) [inline README.md]: typo -> `contructor`

File: README.md [hunk 3]
```diff
@@ -3,3 +3,54 @@
...
+        this.dataAccessLayer = dataAccessLayer;
+    }
+}
+```
+
+## Normal construction
+
+### Example:
+
+*consumer (typicaly a service)*
```
Comment (timofudala) [inline README.md]: typo

File: README.md [hunk 4]
```diff
@@ -3,3 +3,54 @@
...
+```ts
+class Main {
+    private dataAccessLayer: DataAccessLayer;
+
+    constrcutor() {
```
Comment (timofudala) [inline README.md]: typo

**Review by timofudala — CHANGES_REQUESTED** [review]

**Review by Juuldamen — APPROVED** [review]

**Review by timofudala — APPROVED** [review]

**Pull request merged**